Instantiate mutation type when executing it

bzbetty opened the pull request:
potential fix for #126 

Instantiate mutation type when executing it, first via serviceProvider then fallback to Activator.createInstance

also allow registering all mutations from multiple types that match a generic parameter (eg marker interface)

bzbetty commented:
second change in this PR that makes GraphQLMutationAttribute optional (controversial change?) will take any public method, static or instance that is declared directly on the type (not inherited).

lukemurray commented on src/EntityGraphQL/Schema/ISchemaProvider.cs:
```diff
@@ -27,7 +27,7 @@ public interface ISchemaProvider
         SchemaType<TBaseType> AddInputType<TBaseType>(string name, string? description);
         ISchemaType AddInputType(Type type, string name, string? description);
-        void AddMutationsFrom<TType>(TType mutationClassInstance, bool autoAddInputTypes = false) where TType : notnull;
+        void AddMutationsFrom<TType>(bool autoAddInputTypes = false);
```
I try (my best) to avoid breaking changes to the API - trying to follow semver where x.y.z is x - breaking changes, y new features, z fixes. 

Given the change, we could also leave the old methods there and mark them as deprecated. The methods can just do nothing with the object passed in. This wold be a 2.2 release.

Then we can later do a 3 that cleans them up and potentially wraps up more breaking changes.

Thoughts?

bzbetty replied:
Would it work if the mutationClassInstance argument remained but defaults to null?

bzbetty commented:
new commit
* Add mutationClassInstance parameter back in to not break API - also allows user to pass in the mutation class instance if they prefer doing it that way
* added new parameter addNonAttributedMethods which defaults to false to allow methods not marked as mutations to be added but not by default
* reverted some tests to use old api to prove it still works and reduce amount of change
